[vtk] Duplicate dependencies will result in an error

### Operating system

Windows

### Compiler

MSVC

### Steps to reproduce the behavior


My vcpkg have package vtk[qt] and pcl[qt,vtk].
My cmake project has two components that depend on PCL and VTK respectively.
component A 
```
find_package(PCL REQUIRED)
target_link_libraries(${component} PRIVATE ${PCL_LIBRARIES})
```
component B
```
find_package(VTK REQUIRED)
target_link_libraries(${component} PRIVATE ${VTK_LIBRARIES})
```

If I compile the two components at the same time, an error will occur.


### Failure logs

FindPEGL.cmake
```
if(TARGET taocpp::pegtl)
    message(STATUS "Searching for PEGTL - found target taocpp::pegtl")
    set_target_properties(taocpp::pegtl PROPERTIES IMPORTED_GLOBAL TRUE)
    if(NOT TARGET PEGTL::PEGTL)
       add_library(PEGTL::PEGTL IMPORTED INTERFACE)
       target_link_libraries(PEGTL::PEGTL INTERFACE taocpp::pegtl)
    endif()
else()
```

error
```
	"resource": "/C:/src/sfr-vcpkg/installed/x64-windows/share/vtk/FindPEGTL.cmake",
	"owner": "cmake-configure-diags",
	"severity": 8,
	"message": "CMake Error at C:/src/sfr-vcpkg/installed/x64-windows/share/vtk/FindPEGTL.cmake:26 (set_target_properties):Attempt to promote imported target \"taocpp::pegtl\" to global scope (by\nsetting IMPORTED_GLOBAL) which is not built in this directory.",
```


### Additional context

I think it can be modified like this：

```
    get_target_property(TARGET_IMPORTED_GLOBAL taocpp::pegtl IMPORTED_GLOBAL)
      if(NOT TARGET_IMPORTED_GLOBAL)
        set_target_properties(taocpp::pegtl PROPERTIES IMPORTED_GLOBAL TRUE)
    endif()
```

--- a/ports/vtk/pegtl.patch
+++ b/ports/vtk/pegtl.patch
@@ -58,7 +58,7 @@ diff --git a/CMake/FindPEGTL.cmake b/CMake/FindPEGTL.cmake
 index 73eee02f7..22d8bc159 100644
 --- a/CMake/FindPEGTL.cmake	
 +++ b/CMake/FindPEGTL.cmake
-@@ -19,31 +19,43 @@
+@@ -19,31 +19,46 @@
  # Copyright (c) 2009 Benoit Jacob <jacob.benoit.1@gmail.com>
  # Redistribution and use is allowed according to the terms of the 2-clause BSD license.
  
@@ -71,7 +71,10 @@ index 73eee02f7..22d8bc159 100644
 +find_package(PEGTL CONFIG REQUIRED)
 +if(TARGET taocpp::pegtl)
 +    message(STATUS "Searching for PEGTL - found target taocpp::pegtl")
-+    set_target_properties(taocpp::pegtl PROPERTIES IMPORTED_GLOBAL TRUE)
++    get_target_property(TARGET_IMPORTED_GLOBAL taocpp::pegtl IMPORTED_GLOBAL)
++    if(NOT TARGET_IMPORTED_GLOBAL)
++      set_target_properties(taocpp::pegtl PROPERTIES IMPORTED_GLOBAL TRUE)
++    endif()
 +    if(NOT TARGET PEGTL::PEGTL)
 +       add_library(PEGTL::PEGTL IMPORTED INTERFACE)
 +       target_link_libraries(PEGTL::PEGTL INTERFACE taocpp::pegtl)

--- a/ports/vtk/vcpkg.json
+++ b/ports/vtk/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "vtk",
   "version-semver": "9.2.0-pv5.11.0",
-  "port-version": 10,
+  "port-version": 11,
   "description": "Software system for 3D computer graphics, image processing, and visualization",
   "homepage": "https://github.com/Kitware/VTK",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -9126,7 +9126,7 @@
     },
     "vtk": {
       "baseline": "9.2.0-pv5.11.0",
-      "port-version": 10
+      "port-version": 11
     },
     "vtk-dicom": {
       "baseline": "0.8.14",

--- a/versions/v-/vtk.json
+++ b/versions/v-/vtk.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "e887b758153039453976be1e5d0bc54d537b0417",
+      "version-semver": "9.2.0-pv5.11.0",
+      "port-version": 11
+    },
+    {
       "git-tree": "6365f197946995803914141a82c1830d165427b3",
       "version-semver": "9.2.0-pv5.11.0",
       "port-version": 10

